Add recipe for ox-gist

punchagan opened the pull request:
### Brief summary of what the package does

The package makes it easy to share a sub-tree from an `org` file as a `gist` on GitHub. A published `gist` can be easily updated when the contents of the local `org` sub-tree are changed. It uses `defunkt/gist.el` to do the heavy lifting of posting and updating gists on GitHub.

### Direct link to the package repository

https://github.com/punchagan/org2gist

### Your association with the package

I'm the creator/maintainer of this package

### Relevant communications with the upstream package maintainer

None needed

### Checklist

<!-- Please confirm with `x`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thank you for this, and apologies for the delay.  Looks like a cool idea.  A couple very quick comments:

`org2gist.el` with byte-compile using Emacs 27.1:
```
org2gist.el:71:14:Warning: Unknown slot `:html-url'
org2gist.el:71:34:Warning: Unknown slot `:description'
```

`org2gist.el` with [melpazoid](https://github.com/riscy/melpazoid):
- org2gist.el#L80: No `format` required; `message` takes an f-string

I also notice the since instance of `progn` you use isn't necessary for the "else" part of an `if`.

punchagan commented:
@riscy Thanks for taking the time to review my package. Apologies for not submitting something cleaner, from the beginning. I've now fixed the byte-compile warnings and removed the unnecessary `format` and `progn` calls.

riscy commented:
Thank you!  Everything looks great code-wise.  I was thinking another interesting angle for this might be to write this as an org export engine (ox-gist.el or similar).  I believe this is an [example](https://github.com/zzamboni/750words-client/tree/0fed7621c04debad64ea6455455494d4e6eb03fa) of a package that does something similar to that.  Not a blocker if you disagree, but I'll wait to hear your thoughts on this before I merge.  (Otherwise this should be a quick merge.)

punchagan commented:
> Thank you! Everything looks great code-wise. I was thinking another interesting angle for this might be to write this as an org export engine (ox-gist.el or similar). I believe this is an [example](https://github.com/zzamboni/750words-client/tree/0fed7621c04debad64ea6455455494d4e6eb03fa) of a package that does something similar to that. Not a blocker if you disagree, but I'll wait to hear your thoughts on this before I merge. (Otherwise this should be a quick merge.)

This is definitely and interesting idea. I'll try to work on this in the next few days!

punchagan commented:
> Thank you! Everything looks great code-wise. I was thinking another interesting angle for this might be to write this as an org export engine (ox-gist.el or similar).

@riscy Thank you for your suggestion. I have changed the package to be an org export backend `ox-gist.el`and the package now also supports exporting a buffer to a gist, instead of just a sub-tree, which was the case before.  I have run the checks like, `checkdoc`, `package-lint` and `byte-compile-file`. Hopefully, everything should be good with this PR. Let me know if I missed something!

riscy commented:
Looks very cool - thanks for this.

punchagan commented:
> Looks very cool - thanks for this.

@riscy Thank you for the merge, for your helpful suggestions and for guiding me through contributing my first package to melpa.
